### PR TITLE
[SPARK-41637][SQL] ORDER BY ALL

### PR DESCRIPTION
**Pull request by rxin**
### What changes were proposed in this pull request?
This patch adds ORDER BY ALL support to SQL. ORDER BY ALL is a syntactic sugar to sort the output by all the fields, from left to right. It also allows specifying asc/desc as well as null ordering.

### Why are the changes needed?
It's a good convenience sugar added initially by DuckDB to avoid repeating the fields.

### Does this PR introduce _any_ user-facing change?
Yes. See above.

### How was this patch tested?
Added SQL tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveOrderByAll.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          case a: UnresolvedAttribute => a.name.toUpperCase() == ALL
+          case _ => false
+        }
+      }
+    }
+  }
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUpWithPruning(
+    _.containsAllPatterns(UNRESOLVED_ATTRIBUTE, SORT), ruleId) {
+    // This only makes sense if the children is resolved.
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveOrderByAll.scala:69]: nit: child, not children

**Comment by rxin** [comment]
cc @cloud-fan this one too

File: sql/core/src/test/resources/sql-tests/inputs/order-by-all.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
+create temporary view data as select * from values
```
Comment (gengliangwang) [inline sql/core/src/test/resources/sql-tests/inputs/order-by-all.sql:1]: nit: we can have a test case combining `group by all` and `order by all`. It should work, just for covering a testing scenario.

**Review by gengliangwang — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/inputs/order-by-all.sql (hunk 2 above)
Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/order-by-all.sql:1]: isn't it to better to test different features in test golden files?

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/inputs/order-by-all.sql (hunk 2 above)
Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/inputs/order-by-all.sql:1]: Oh you mean test both feature in one query? That's a good idea, we can add later.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**